packages/blockchain-extension/extension/commands/gatewayConnectCommand.ts

```diff
@@ -41,16 +43,26 @@ export async function gatewayConnect(gatewayRegistryEntry: FabricGatewayRegistry
 
     const gatewayName: string = gatewayRegistryEntry.displayName ? gatewayRegistryEntry.displayName : gatewayRegistryEntry.name;
 
-    // TODO JAKE: Update this so it supports managed ansible environments
-    // Maybe we'll want to use the 'managedGateway' property here. We'll need to find out which environment this gateway is for as well.
-    if (gatewayName.includes(`${FabricRuntimeUtil.LOCAL_FABRIC} - `)) {
-        const running: boolean = await LocalEnvironmentManager.instance().getRuntime().isRunning();
+    const environmentName: string = gatewayRegistryEntry.fromEnvironment;
+    let environmentEntry: FabricEnvironmentRegistryEntry;
+    if (environmentName) {
+        environmentEntry = await FabricEnvironmentRegistry.instance().get(environmentName);
+    }
+
+    if (environmentEntry && environmentEntry.managedRuntime && environmentEntry.environmentType === EnvironmentType.ANSIBLE_ENVIRONMENT) {
```

Comment:
do you really need to check the environment type here since all managed runtimes are ansible ones?

Reply:
Changed.